var/spack/repos/builtin/packages/slurm/package.py

```diff
@@ -0,0 +1,103 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+from spack import *
+
+
+class Slurm(AutotoolsPackage):
+    """Slurm is an open source, fault-tolerant, and highly scalable cluster
+    management and job scheduling system for large and small Linux clusters.
+
+    Slurm requires no kernel modifications for its operation and is relatively
+    self-contained. As a cluster workload manager, Slurm has three key
+    functions. First, it allocates exclusive and/or non-exclusive access to
+    resources (compute nodes) to users for some duration of time so they can
+    perform work. Second, it provides a framework for starting, executing,
+    and monitoring work (normally a parallel job) on the set of allocated
+    nodes. Finally, it arbitrates contention for resources by managing a
+    queue of pending work.
+    """
+
+    homepage = 'https://slurm.schedmd.com'
+    url = 'https://github.com/SchedMD/slurm/archive/slurm-17-02-6-1.tar.gz'
+
+    version('17-02-6-1', '8edbb9ad41819464350d9de013367020')
+
+    variant('gtk', default=False, description='Enable GTK+ support')
+    variant('mariadb', default=False, description='Use MariaDB instead of MySQL')
+
+    variant(
+        'features',
+        values=('hwloc', 'hdf5'),
+        default='hwloc',
+        multi=True,
+        description='Enable a few selected features of Slurm'
+    )
```

Comment:
Hmm, I feel like this would be more natural as two separate variants. In my mind, multi-valued variants are useful when things are mutually exclusive, or when multiple values are allowed. If it _can_ be done with 2 Boolean variants without any need for `conflicts()`, then it should be done that way.